x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_exploration/components/outlier_exploration/use_outlier_data.ts

```diff
@@ -81,12 +82,16 @@ export const useOutlierData = (
 
   const fetchColumnChartsData = async function () {
     try {
-      if (jobConfig !== undefined) {
-        const columnChartsData = await fetchChartsData(
-          jobConfig.dest.index,
-          ml.esSearch,
-          searchQuery,
-          columns.filter((cT) => dataGrid.visibleColumns.includes(cT.id))
+      if (jobConfig !== undefined && indexPattern !== undefined) {
+        const dataLoader = new DataLoader(indexPattern, getToastNotifications());
```

Comment:
Please avoid usting cached `getToastNotifications` and retriveve `toastNotifications` from the kibana context.

Reply:
The rest of the file uses the same approach and I didn't plan to refactor it in this PR.